includes/view/ShiftCalendarShiftRenderer.php

```diff
@@ -173,63 +174,43 @@ private function renderShiftNeededAngeltype(Shift $shift, $shift_entries, $angel
-                // No link and add a text hint, when the shift ended
-                $entry_list[] = $inner_text . ' (' . __('ended') . ')';
-                break;
-
-            case ShiftSignupState::NOT_ARRIVED:
-                // No link and add a text hint, when the shift ended
-                $entry_list[] = $inner_text . ' (' . __('please arrive for signup') . ')';
-                break;
-
-            case ShiftSignupState::NOT_YET:
-                $entry_list[] = $inner_text . ' (' . __('not yet') . ')';
-                break;
-
-            case ShiftSignupState::ANGELTYPE:
-                if ($angeltype->restricted) {
-                    // User has to be confirmed on the angeltype first
-                    $entry_list[] = $inner_text . icon('mortarboard-fill');
-                } else {
-                    // Add link to join the angeltype first
-                    $entry_list[] = $inner_text . '<br />'
-                        . button(
-                            page_link_to(
-                                'user_angeltypes',
-                                ['action' => 'add', 'angeltype_id' => $angeltype->id]
-                            ),
-                            sprintf(__('Become %s'), $angeltype->name),
-                            'btn-sm'
-                        );
-                }
-                break;
-
-            case ShiftSignupState::COLLIDES:
-            case ShiftSignupState::SIGNED_UP:
-                // Shift collides or user is already signed up: No signup allowed
-                $entry_list[] = $inner_text;
-                break;
-
-            case ShiftSignupState::OCCUPIED:
-                // Shift is full
-                break;
+        $entry = match ($shift_signup_state->getState()) {
+            // When admin or free display a link + button for sign up
+            ShiftSignupStatus::ADMIN, ShiftSignupStatus::FREE =>
+                '<a class="me-1 text-nowrap" href="'
+                . shift_entry_create_link($shift, $angeltype)
+                . '">'
+                . $inner_text
+                . '</a> '
+                . button(
+                    shift_entry_create_link($shift, $angeltype),
+                    __('Sign up'),
+                    'btn-sm btn-primary text-nowrap d-print-none'
+                ),
+            // No link and add a text hint, when the shift ended
+            ShiftSignupStatus::SHIFT_ENDED => $inner_text . ' (' . __('ended') . ')',
+            // No link and add a text hint, when the shift ended
+            ShiftSignupStatus::NOT_ARRIVED => $inner_text . ' (' . __('please arrive for signup') . ')',
+            ShiftSignupStatus::NOT_YET => $inner_text . ' (' . __('not yet') . ')',
+            ShiftSignupStatus::ANGELTYPE => $angeltype->restricted ?
+                // User has to be confirmed on the angeltype first
+                $inner_text . icon('mortarboard-fill')
+                :
```

Comment:
I would prefer having the `?` and `:` at the start of the statements instead in a new line to make it clearer.